activerecord/lib/active_record/runtime_registry.rb

```diff
@@ -25,15 +25,57 @@ def async_sql_runtime=(runtime)
       ActiveSupport::IsolatedExecutionState[:active_record_async_sql_runtime] = runtime
     end
 
+    def queries_count
+      ActiveSupport::IsolatedExecutionState[:active_record_queries_count] ||= 0
+    end
+
+    def queries_count=(count)
+      ActiveSupport::IsolatedExecutionState[:active_record_queries_count] = count
+    end
+
+    def cached_queries_count
+      ActiveSupport::IsolatedExecutionState[:active_record_cached_queries_count] ||= 0
+    end
+
+    def cached_queries_count=(count)
+      ActiveSupport::IsolatedExecutionState[:active_record_cached_queries_count] = count
+    end
+
     def reset
+      reset_runtimes
+      reset_queries_count
+      reset_cached_queries_count
+    end
+
+    def reset_runtimes
       rt, self.sql_runtime = sql_runtime, 0.0
       self.async_sql_runtime = 0.0
       rt
     end
+
+    def reset_queries_count
+      qc = queries_count
+      self.queries_count = 0
+      qc
+    end
+
+    def reset_cached_queries_count
+      qc = cached_queries_count
+      self.cached_queries_count = 0
+      qc
+    end
   end
 end
 
 ActiveSupport::Notifications.monotonic_subscribe("sql.active_record") do |name, start, finish, id, payload|
+  unless ["SCHEMA", "TRANSACTION"].include?(payload[:name])
+    if payload[:cached]
+      ActiveRecord::RuntimeRegistry.cached_queries_count += 1
+    else
+      ActiveRecord::RuntimeRegistry.queries_count += 1
```

Comment:
I wonder if it makes more sense to include `cached_queries` in `queries`.

Reply:
Do you mean 1) to not have a separation of "cached" and "not cached" or 2) to include cached into uncached?

Reply:
Yeah, that was exactly my intention originally, but made a bug in the implementation 🤦.
Updated with the suggestion, thank you! ❤️